src/rendering/SceneManager.cc

```diff
@@ -258,16 +258,11 @@ rendering::VisualPtr SceneManager::CreateVisual(Entity _id,
     /// In general, this can be used to store any local transforms between the
     /// parent Visual and geometry.
     rendering::VisualPtr geomVis;
-    if (localPose != math::Pose3d::Zero)
```

Comment:
Most visuals don't have a local pose. I think only the planes do. So removing this check is essentially duplicating the number of visuals in the scene by nesting an extra visual inside every visual. It may be worth it looking into performance impacts.

Another thing to keep in mind is that there may be code assuming that the visual with the entity ID has a geometry, as opposed to having a child visual. So I think this has the potential to break existing plugins and other user code.

Reply:
just ran into an issue with this new node structure. This unfortunately broke thermal cameras which relies on the entity ID stored in the geometry to point to the parent visual but now it's pointing to a geometry visual.

Reply:
Okay, would it be better to add a local pose member to `rendering::Geometry`? Do we have other alternatives?

Reply:
hmm how about we do this for the ground plane as special case? i.e.

```
    if (localPose != math::Pose3d::Zero)
    {
      rendering::VisualPtr geomVis = this->dataPtr->scene->CreateVisual(name + "_geom");
      geomVis->SetLocalPose(localPose);
      geomVis->AddGeometry(geom);
      visualVis->AddChild(geomVis);
    }
    else
    {
      visualVis->AddGeometry(geom);
    }
    visualVis->SetLocalScale(scale);
```

Reply:
> hmm how about we do this for the ground plane as special case? i.e.

Yup, I can do that, but it would affect all plane shapes, not just the ground plane. This could still break existing code though if, as @chapulina said, the code assumes that the visual with the entity ID always has a geometry, as opposed to having a child visual.

Reply:
Changed in https://github.com/ignitionrobotics/ign-gazebo/pull/574/commits/d3c07322154026610eba39ad7c22f7a4df46c9a9

Reply:
yeah.. I can't think of a good workaround for now. Adding `localPose` to Geometry in ign-rendering also means internally we'll be creating a node to store the geometry and apply transformation to this parent node.

One approach I'm thinking of is to transform the vertices of the plane that gets created in ign-rendering based on its normal. Alternatively add an Transform function ign-common's Mesh class to we can rotate it before feeding the mesh data to ign-rendering.

Reply:
I tried the current fix and it didn't seem to break any GUI functionality, like selection, highlight and collision visualization. So at least our plugins seem to be ok with the change. There's a small chance that this breaks downstream plugins, but I think the chance is small enough that the benefits outweigh it. Let's get this in!